sencha/translate.py

```diff
@@ -304,30 +332,38 @@ def check_nucleotide_content(self, description, n_kmers, sequence):
         pass thresholds to be
         checked for protein-coding-ness
         """
+        fasta_seqs = {"low_complexity_nucleotide": []}
         if n_kmers > 0:
             jaccard = np.nan
             special_case = "Low complexity nucleotide"
-            self.maybe_write_fasta(
-                self.file_handles["low_complexity_nucleotide"], description, sequence
-            )
+            fasta_seqs["low_complexity_nucleotide"].append([description, sequence])
         else:
             jaccard = np.nan
             n_kmers = np.nan
             special_case = "Read length was shorter than 3 * peptide " "k-mer size"
-        return [
+        tranlated_scores = [
             constants_translate.SingleReadScore(jaccard, n_kmers, special_case, i)
             for i in [1, 2, 3, -1, -2, -3]
         ]
+        lines = [
+            self.get_coding_score_line(description, x, y, z, frame)
+            for x, y, z, frame in tranlated_scores
+        ]
+        return lines, fasta_seqs
 
-    def maybe_score_single_read(self, description, sequence):
+    def maybe_score_single_read(self, record):
```

Comment:
is record also a named tuple? if not i think a named tuple would also be really nice here.

Reply:
record a screeddb object with seq description and name